[XWI-60] Documentation of admin guide (#23884)
opf/openproject: docs/system-admin-guide/wikis/wiki-providers/README.md

---
sidebar_navigation:
  title: Wiki providers
  priority: 100
description: External wiki providers for OpenProject
keywords: wikis, xwiki, wiki integration, wiki provider, external wiki
---

# Wiki providers

You can connect your OpenProject instance to external wiki providers. To do so, navigate to *Administration → Wikis →
Wiki providers*. Here all configured wiki providers are listed. With the *+ Wiki provider* button, you can add a new

> wiki provider from the list of supported instances.

Reviewer: The "supported instances" threw me off, because we don't support specific instances of a wiki, but rather specific... types... kinds... brands... APIs...

Not really sure which word I'd prefer, but "instance" sounds weird right now.

Reviewer 2: I agree, instances maybe isnt the most clear word here. How about "options"?

Author: Went for options for now.

> *Insert a screenshot from the index page*

Once a wiki provider is configured, it will be available globally in the whole OpenProject instance.

| Topic                                                              | Description                                      |
|--------------------------------------------------------------------|:-------------------------------------------------|
| [XWiki](#xwiki)                                                    | Section about configuration of XWiki providers   |
| [XWiki: Set up](#set-up-the-xwiki-provider)                        | How to set up an XWiki provider                  |
| [XWiki: Replacing authentication](#replace-authentication-details) | Replacing authentication details and its effects |

> | [Deleting a wiki provider](#deleting-a-wiki-provider)              | Effects of removing a wiki provider              |

Reviewer 2: ```suggestion
| [Delete a wiki provider](#delete-a-wiki-provider)              | Effects of removing a wiki provider              |
```

## XWiki

[feature: xwiki_integration]

OpenProject offers an integration with XWiki to allow users to:

- Link and create XWiki pages inside rich text areas
- Link and create XWiki pages as relations to work packages
- Link work packages as project management relations to an XWiki page
- Link and create work packages from inside XWiki page content

### Set up the XWiki provider

> [!IMPORTANT]

> > You need to have administrator privileges in your XWiki and OpenProject instances to set up this integration.

Reviewer 2: ```suggestion
> You need to have administrator privileges both in your XWiki and OpenProject instances to set this integration up.
```

#### 1. Install the OpenProject plugin in XWiki

> Start by opening the XWiki instance as an administrator. Click on the *Drawer* and select *Administer Wiki*. Click on

Reviewer 2: I don't think I've ever seen the term *Drawer* in our documentation. Can we say "Navigation menu in top right corner" instead? 
That would be Screenshot 1. 
Extension - Screenshot 2. 
Search and install - Screenshot 3. Though on our instance it's already been installed, so it will not show the installation button. I think it's fine for now.

Author: Well, I didn't come up with that name, this is how XWiki calls their menu:

<img width="355" height="120" alt="image" src="https://github.com/user-attachments/assets/9621ab86-acfd-4cf1-92ac-cf837ca6e414" />

Author: But I could use "Burger menu" or "Hamburger menu" or something equally accepted as that.

*Extensions*, search for **OpenProject Integration (Pro)**, and install the plugin.

> *How many screenshots are needed here?*

#### 2. Create the XWiki provider in OpenProject

> Navigate to the OpenProject administration settings page. In the side menu on the left select *Wikis → Wiki providers*.

Reviewer 2: ```suggestion
Navigate to the OpenProject administration settings page. In the left hand menu select *Wikis → Wiki providers*.
```

> Click on the **+ Wiki provider** button and select **XWiki**.

Reviewer 2: ```suggestion
Click the **+ Wiki provider** button and select **XWiki**.
```

> *Screenshot of creation button*

A new page titled **New XWiki provider** will appear, where you will be able to configure your XWiki instance.

> First, choose a **Name** for the XWiki provider. This name will show up when linking pages from this XWiki instance in

Reviewer 2: ```suggestion
First, choose a **Name** for the XWiki provider. This name will be shown when linking pages from this XWiki instance in
```

OpenProject.

Then, enter the **Instance URL** of your XWiki instance. This URL will be checked on submitting this form.

> Last, choose the **Authentication method**. This determines how OpenProject user will connect their user accounts to

Reviewer 2: ```suggestion
Lastly, choose the **Authentication method**. This determines how OpenProject users will connect their user accounts to
```

their XWiki accounts. Additional information is provided in the next step.

> *Screenshot of the fist step*

#### 3. Configure authentication credentials

Depending on which authentication method you chose in the previous step, this configuration step differs. Currently,

> OpenProject supports the following authentication methods:

Reviewer: While I (as a software developer) like the preparation for adding more authentication methods in the future, this part of the docs reads a bit mechanical to me:

> Chose from any of the numbers below
>
> * 1

Should we with one or two words acknowledge, that right now there's just one legal auth method?

Reviewer 2: +1

Author: rephrased

Reviewer: good enough for me.

- **Two-way OAuth 2.0 authorization code flow**

##### Two-way OAuth 2.0 authorization code flow

This method requires the user to go through an OAuth 2.0 authorization code flow to connect their OpenProject user
account to their XWiki account, and vice versa.

First, you will see the client credentials generated by OpenProject. Copy the **Application ID** and **Application
secret** to your clipboard and insert them in the corresponding form at the XWiki instance. A link to the correct
configuration page is displayed in a banner above.

> *Screenshot of the first state of the authentication step*

Once this is done, click on **Done, continue** and the next step is shown, where a **Client ID** and **Client secret**

> are generated to be used for the XWiki OAuth client configuration. Those values are just examples – if needed, you can

Reviewer 2: ```suggestion
are generated to be used for the XWiki OAuth client configuration. These values are just examples – if needed, you can
```

> change them. The generated secret fulfills all requirements to be considered secure. Copy the all values, including the

Reviewer: Typo. Either

```suggestion
change them. The generated secret fulfills all requirements to be considered secure. Copy the values, including the
```

or

```suggestion
change them. The generated secret fulfills all requirements to be considered secure. Copy all the values, including the
```

> **Redirect URI**, to your clipboard and insert them in the corresponding form at the XWiki instance. A link will be

Reviewer: Maybe we'll need to show how to reach the correct form, no?

IIRC this was the step where you have to select "OAuth client" from an extra long select box, before you arrive at the right form.

Author: I usually would discourage full configuration pathes in external applications. If XWiki changes their UI, our documentation is outdated in no time. We did the same for Azure and SharePoint. We could though mention, that we have a link in our own UI?

Reviewer: I think the link in our UI is not deep enough. It leads you here, where you have to find the thing in a long list:

[Screencast from 2026-07-01 10-38-22.webm](https://github.com/user-attachments/assets/944df22b-8b00-4d44-9673-8e7a1683298b)

And no, the "New ... object" link is not usable, because it's only there **after** you added the first object of this kind.

Author: Talked to Maya and she agrees, that we should keep our side of the doc(k)s more general. We could try to improve the link, though.

displayed again above the form.

> *Screenshot of the second state of the authentication step*

Once this is done, click on **Save and continue**. This will conclude the configuration of the XWiki provider and the
details page will be shown.

### Replace authentication details

If authentication credentials are lost, leaked, or need to be rerolled, you can replace them by clicking on the name of

> an XWiki provider. This will open the details page of the provider. Click on the **Replace** next to the configuration

Reviewer 2: ```suggestion
an XWiki provider. This will open the details page of the provider. Click **Replace** next to the configuration
```

entry. The replacement action is available separately for each pair of credentials and will generate new credentials
after confirming the destructive action. As described in the
section [Two-way OAuth 2.0 authorization code flow](#two-way-oauth-20-authorization-code-flow) you will have to copy the
new values over to the corresponding XWiki forms.

> [!IMPORTANT]
> Every user will have to reconnect from the affected side after resetting the credentials. For example, if replacing

> > the Wiki OAuth credentials, the connection from the OpenProject account to the XWiki account is lost and has to be

Reviewer: Did you confirm that this is true?

I know that we exchange the entire OAuth application in OpenProject, but this is not technically neccessary. I am pretty sure that you can exchange client id and secret in XWiki without throwing away the entire client, so I'd expect that the re-connect is not necessary in this case.

P.S.: I am still open to discuss whether OpenProject should support credential rotation without disconnecting every user.

Author: Just confirmed it. Both sides remove the access tokens. 

TBH, I don't even remember the exact reasoning, but it was called a "best practice". If credentials are invalidated - and that's what happens to the old ones - you should invalidate all access tokens given to clients. Maybe we can do some research on that topic.

Author: I reread your statement and am stumbling across: 

> I am pretty sure that you can exchange client id [...] without throwing away the entire client

what? if you change the id, it is a different client, no? Don't think in "database entity", think in "I am allowed to connect to a application with this credentials". So, it is not an exchange of two values inside the same client. It is an invalidation of a previous one and a creation of a new one.

Reviewer: I am absolutely thinking in database entities, or to be more specific: in UI that is offered.

I just realize that you are right still, because we throw away the client on our side, so even if XWiki keeps the same stuff around, we'll have thrown away all user tokens that we obtained.

However:

> So, it is not an exchange of two values inside the same client. It is an invalidation of a previous one and a creation of a new one.

This is purely a decision on our end. Imagine you had to create a new account at any online service if you wanted to change your password. This would be nuts. And that's what we do right now, when we force the creation of a new client just to rotate the client secret.

I can kinda follow your argument on the client id. At least a bit. But I still would disagree. Of course it should be possible to change details about a client without throwing everything away. Maybe the client was called `openproject` back in the day, but nowadays you have two OP instances and it would be cool to call them `openproject-a` and `openproject-b` for clarity.

Needless to say that we steer around this problem elegantly by forcing users to use unreadable client ids by default.

Author: But that's just it, right? You can change easily your password, but NOT you user id. Sometimes services allow to change your login, but only jumping through multiple hoops of reauthenticating.

We change the full set, id and secret, this is IMHO the equivalent to creating a new one. There is no way of only changing the secret.

Reviewer: As I said, I just realized that we drop the entire client and tokens anyways.

But that's totally a decision that we made. We don't **have to** do any of this.

(and yes, changing usernames is often easily possible... many services identify you by your email address and there it's obviously possible; I should also mention that all of the above can be changed in Keycloak)

Author: > many services identify you by your email address and there it's obviously possible

yes, but how many do you know, where you do not have to reauthenticate after doing so?

Reviewer: I love to keep on discussing ^^

Can we agree that there is a difference between asking the user that changed their credentials to log in again versus asking all users to log in again, when their admin changes a setting that they don't know about?

Anyways: I'll point out once more, that the documentation you proposed here is in fact correct xD

Author: > Can we agree that there is a difference between asking the user that changed their credentials to log in again versus asking all users to log in again, when their admin changes a setting that they don't know about?

agreed, I only tried to reason with myself about when and why to invalidate sessions.

Anything left here to do?

> reconnected.

> ## Deleting a wiki provider

Reviewer 2: ```suggestion
## Delete a wiki provider
```

Reviewer 2: The rest of the headings use active verb forms, I'd keep it consistent

You can delete a wiki provider by clicking on the provider's name in the list. This will open the details page of the

> provider. In here, you click on the **Delete** button in the top right corner.

Reviewer 2: ```suggestion
provider. There you can click the **Delete** button in the top right corner.
```

> Deleting a wiki provider will remove all wiki page links created as relation for a work package. In addition, all wiki

Reviewer 2: ```suggestion
> [!IMPORTANT]
> Deleting a wiki provider will remove all wiki page links created as relation for a work package. In addition, all wiki page links that are mentioned in rich text areas will show an error, as OpenProject will no longer be able to fetch information about the linked wiki page.
```

Author: done

> page links that are mentioned in rich text areas will show an error, as OpenProject will no longer be able to fetch

Reviewer 2: ```suggestion
```

> information about the linked wiki page.

Reviewer 2: ```suggestion
```
